### PR TITLE
Feature web appointment recommendation

### PR DESCRIPTION
**Pull request by VujicGit**

### CODE REVIEW

**Review by st3fan7 — APPROVED** [review]

File: Backend/Schedules/Service/AvailableAppointmentService.cs [hunk 1]
```diff
@@ -191,6 +191,6 @@ public void SwitchStrategy(PriorityType priority)
 
         public IAppointmentRepository appointmentRepository;
         public WorkDayService workDayService;
-        public IPriorityStrategy strategy;
+        public IPriorityStrategy strategy;*/
     }
 }
```
Comment (jelena-vlajkov) [inline Backend/Schedules/Service/AvailableAppointmentService.cs:196]: If the entire class is commented, please delete it.

File: Backend/Schedules/Service/StrategyPattern/DatePriorityStrategy.cs [hunk 2]
```diff
@@ -7,87 +7,64 @@
...
-            return RecommendResults(allAppointmentsInTimePeriod, parameters);
-        }
+    public class DatePriorityStrategy : IPriorityStrategy
+    {
+        private IAppointmentService _appointmentService;
+        private IDoctorService _doctorService;
 
-        public Appointment RecommendResults(List<Appointment> allAppointmentsInTimePeriod, PriorityParameters parameters)
+        private const int NUMBER_OF_RECOMMENDED_APPOINTMENTS = 5;
+        public DatePriorityStrategy(IAppointmentService appointmentService, IDoctorService doctorService)
```
Comment (jelena-vlajkov) [inline Backend/Schedules/Service/StrategyPattern/DatePriorityStrategy.cs:24]: Find a way to make number more configurable.

File: Backend/Users/Model/Specialization.cs [hunk 3]
```diff
@@ -16,23 +17,25 @@ public class Specialization : IIdentifiable<int>
...
             return Id;
-        }
-
+        }
+
+        /*[ForeignKey("Doctor")]
+        public string DoctorId { get; set; }
+        public virtual Doctor Doctor { get; set; }*/
+
         public void SetId(int id)
```
Comment (jelena-vlajkov) [inline Backend/Users/Model/Specialization.cs:35]: Delete commented lines.

File: Backend/Users/Model/Doctor.cs [hunk 4]
```diff
@@ -28,12 +29,15 @@ public class Doctor : Employee
...
+        public int SpecializationId { get; set; }
+
+        public virtual Specialization Specialization { get; set; }
 
 
         public Doctor() 
         {
-            Specializations = new List<Specialization>();
+            //Specializations = new List<Specialization>();
         }
```
Comment (jelena-vlajkov) [inline Backend/Users/Model/Doctor.cs]: Delete commented lines.

File: Backend/Users/Repository/FileRepository/UserRepository.cs [hunk 5]
```diff
@@ -235,5 +220,10 @@ public IEnumerable<Doctor> GetDoctorsFromDepartment(Department department)
             var doctors = GetAllDoctors().Where(entity => entity.Department != null);
             return doctors.Where(entity => entity.Department.Id == department.Id);
         }
+
+        public IEnumerable<Doctor> GetAllDoctorsBySpecialization(Specialization specialization)
+        {
+            throw new NotImplementedException();
+        }
     }
```
Comment (jelena-vlajkov) [inline Backend/Users/Repository/FileRepository/UserRepository.cs:228]: Never leave a method unimplemented.

File: Backend/Users/Repository/MySqlRepository/DoctorSqlRepository.cs [hunk 6]
```diff
@@ -74,6 +72,6 @@ public Doctor Update(Doctor entity)
             _context.Doctors.Update(entity);
             _context.SaveChanges();
             return entity;
-        }
+        }*/
     }
```
Comment (jelena-vlajkov) [inline Backend/Users/Repository/MySqlRepository/DoctorSqlRepository.cs:35]: Delete commented lines.

File: UnitTests/Examinations/ReportTests.cs [hunk 7]
```diff
@@ -126,7 +126,7 @@ private static Doctor CreateDoctor()
                 DepartmentId = 1,
                 ExaminationRoomId = 1,
                 OperationRoomId = 2,
-                Specializations = new List<Specialization>()
+                //Specialization = new Specialization(1, "Interna medicina")
             };
```
Comment (jelena-vlajkov) [inline UnitTests/Examinations/ReportTests.cs:130]: Delete commented lines.

**Review by jelena-vlajkov — DISMISSED** [review]
Great algorithm! 
Just delete the all commented lines and implement the method that's not implemented.

**Review by st3fan7 — CHANGES_REQUESTED** [review]
Resolve conflict.

**Review by oakenknight — APPROVED** [review]

**Review by st3fan7 — APPROVED** [review]

**Pull request merged**